gridstatus/isone_api/isone_api.py

```diff
@@ -382,3 +341,201 @@ def get_dayahead_hourly_demand(
+
+        elif horizons == "all":
+            url = f"{BASE_URL}/hourlyloadforecast/all/day/{date.strftime('%Y%m%d')}"
+        else:
+            url = f"{BASE_URL}/hourlyloadforecast/day/{date.strftime('%Y%m%d')}"
+
+        response = self.make_api_call(url)
+        df = pd.DataFrame(response["HourlyLoadForecasts"]["HourlyLoadForecast"])
+        return self._handle_load_forecast(df, interval_minutes=60)
+
+    @support_date_range("DAY_START")
+    def get_reliability_region_load_forecast(
+        self,
+        date: str | pd.Timestamp = "latest",
+        end: str | pd.Timestamp | None = None,
+        horizons: Literal["latest", "all"] = "all",
+        verbose: bool = False,
+    ) -> pd.DataFrame:
+        """
+        Get the regional load forecast data for specified date range and horizons.
+
+        Args:
+            date (str): The start date for the data request. Use "latest" for most recent data.
+            end (str | None): The end date for the data request. Only used if date is not "latest".
+            horizons (Literal["latest", "all"]): The horizon for the data request. Options are "latest" or "all".
+
+        Returns:
+            pandas.DataFrame: A DataFrame containing the regional load forecast data for all requested locations.
+        """
+
+        if date == "latest":
+            url = f"{BASE_URL}/reliabilityregionloadforecast/current"
+        elif horizons == "all":
+            url = f"{BASE_URL}/reliabilityregionloadforecast/day/{date.strftime('%Y%m%d')}/all"
+        else:
+            url = f"{BASE_URL}/reliabilityregionloadforecast/day/{date.strftime('%Y%m%d')}"
+
+        response = self.make_api_call(url)
+        df = pd.DataFrame(
+            response["ReliabilityRegionLoadForecasts"]["ReliabilityRegionLoadForecast"],
+        )
+        return self._handle_load_forecast(df, interval_minutes=60)
+
+    def _handle_load_forecast(
+        self,
+        df: pd.DataFrame,
+        interval_minutes: int = 60,
+    ) -> pd.DataFrame:
+        """
+        Process load forecast DataFrame: convert types, rename columns, and add Interval End.
+
+        Args:
+            df (pd.DataFrame): Input DataFrame with load forecast data.
+            interval_minutes (int): Duration of each interval in minutes. Default is 60.
+
+        Returns:
+            pd.DataFrame: Processed DataFrame.
+        """
+        try:
+            # Try the standard pandas datetime conversion first
+
+            date_columns = ["BeginDate", "CreationDate"]
+            df[date_columns] = df[date_columns].apply(pd.to_datetime)
+            df[date_columns] = df[date_columns].apply(
+                lambda x: x.dt.tz_convert(self.default_timezone),
+            )
+
+        except AttributeError:
+            # NOTE(kladar) consistently the above logic fails for DST conversion days.
+            # This catches those and fixes it.
+            # Data coming out looks good, no missed intervals and no duplicates.
```

Comment:
There should be a missed interval when DST starts right? Because the 2am local hour doesn't exist.

Reply:
Anyway, this looks right

```python
ISONEAPI().get_hourly_load_forecast('2024-03-10')['Interval Start'].unique()[:4]

['2024-03-10 00:00:00-05:00', '2024-03-10 01:00:00-05:00', '2024-03-10 03:00:00-04:00', '2024-03-10 04:00:00-04:00']
Length: 4, dtype: datetime64[ns, US/Eastern]
```